src/DeepLinkModule/javasource/deeplink/actions/StartDeeplinkJava.java

```diff
@@ -171,7 +171,7 @@ private void serveDeeplink(String[] args, IMxRuntimeRequest request, IMxRuntimeR
 			
 			if (deeplinkObj == null) {
 				StartDeeplinkJava.logger.warn("Deeplink with name '" + args[2] + "' not found. ");
-				serve404(request, response, session == null ? null : session.getUser().getName());
+				serve404(request, response, session == null ? null : session.getUser(context).getName());

```

Comment:
The `context` here is created using `Core.createSystemContext`, which doesn't have a user attached. Probably we need to use `session.createContext`. (Many places)